test/functionalTest/cases/2549_append_or_update_in_actionType_in_notifications/append_or_update_in_actionType_in_notifications.test

```diff
@@ -0,0 +1,328 @@
+# Copyright 2016 Telefonica Investigacion y Desarrollo, S.A.U
+#
+# This file is part of Orion Context Broker.
+#
+# Orion Context Broker is free software: you can redistribute it and/or
+# modify it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# Orion Context Broker is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
+# General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with Orion Context Broker. If not, see http://www.gnu.org/licenses/.
+#
+# For those usages not covered by this license please contact with
+# iot_support at tid dot es
+
+# VALGRIND_READY - to mark the test ready for valgrindTestSuite.sh
+
+--NAME--
+append or update in actionType in notifications
+
+--SHELL-INIT--
+dbInit CB
+brokerStart CB
+accumulatorStart --pretty-print
+
+--SHELL--
+
+#
+# 01. Create subscription for E1/A1
+# 02. Create E1/A1
+# 03. See 'append' in notification and reset
+# 04. Update A1 using PUT /v2/entities/E1/attrs/A1/value
+# 05. See 'update' in notification and reset
+# 06. Update A1 using POST /v2/entities/E1/attrs
+# 07. See 'update' in notification and reset
+# 08. Add A2 using POST /v2/entities/E1/attrs
+# 09. See 'append' in notification
+#
```

Comment:
I'd suggest to add a new step to check that that a "mixed" situation works. In particular:

* Do a `POST /v2/entities/E1/attrs` including both A1 (existing) and A3 (not existing)
* Check that actionType is update for A1 and append for A3